dist/react-hammerjs.es.js

```diff
@@ -149,7 +149,10 @@ var HammerComponent = function (_React$Component) {
 
 			var self = this;
 			props.ref = function (domElement) {
-				if (self.props.ref) {
+				if (self.props.children && self.props.children.ref) {
```

Comment:
I'm not sure, whether this is the best solution because `ref` is a [special prop](https://reactjs.org/warnings/special-props.html).
Actually, the problem is not only that the child's ref is swallowed but also that `self.props.ref` will always return `undefined` (and I would expect the same for `self.props.children.ref`). A `ref` callback added to the `<Hammer>` component will still be executed (but independently, i.e. it will never be called here and if it did, it would be executed twice).

Maybe it is possible to add another prop to the official interface of the Hammer component that will be executed here. Something like 

```xml
<Hammer
    onPan={onPanHandler}
    refCallback={domElement => myRef = domElement /* domElement will be the child, i.e. the div below */}
>
    <div>something cool</div>
</Hammer>
```

and then

```javascript
props.ref = function(domElement) {
    if (self.props.refCallback) {
        self.props.refCallback(domElement);
    }
    self.domElement = domElement;
}
```

This would still swallow the child's ref but if it's documented that `refCallback` of the Hammer component will get passed the actual child element, one can just place the child's ref there and it will work as expected. This would also avoid trying to access the ref prop directly.

Reply:
I second using a unique prop name such as `refCallback` as to avoid conflict with React special props.